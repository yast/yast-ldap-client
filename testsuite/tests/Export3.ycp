/**
 *  Read.ycp
 *  Test of Ldap:Read function - test more KDC's (bnc#729174)
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    // testedfiles: Ldap.ycp
    import "Testsuite";

    map READ = $[
	"etc": $[
	    "nsswitch_conf" : $[
		"passwd"	: "sss",
		"group"		: "sss",
		"passwd_compat"	: nil,
		"group_compat"	: nil
	    ],
	    "ldap_conf" : $[
		"v" : $[
		    "/etc/ldap.conf": $[
			"host" : "localhost",
			"base" : "dc=suse,dc=cz",
			"nss_base_passwd" : nil,
			"nss_base_shadow" : nil,
			"nss_base_group" : "ou=group,dc=suse,dc=cz",
			"ldap_version": nil,
			"ssl": nil,
			"pam_password": "crypt",
			"tls_cacertdir"	: "/etc/openldap/cacerts/",
			"tls_cacertfile": nil,
			"tls_checkpeer" : nil,
			"uri"		: nil
		    ]
		]
	    ],
	    "krb5_conf" : $[
		"v" : $[
		    "libdefaults": $[
			"default_realm": ["SUSE.CZ"],
		    ],
		    "SUSE.CZ": $[
			"kdc": ["kdc.suse.cz", "kdc.suse.de"],
		    ],
		],
	    ],
	    // /etc/security/pam_*
	    "security": $[
		"section": $[
		    "/etc/security/pam_unix2.conf": $[],
		],
		"v": $[
		    "/etc/security/pam_unix2.conf": $[
			"auth"	: ""
		    ],
		]
	    ],
	    "sssd_conf"	: $[
		"v"			: $[
		    "domain/default"	: $[
			"krb5_realm"	: "SUSE.CZ",
			// this ensures kdc is read from krb5.conf
			"krb5_kdcip"	: nil,
			"ldap_schema"	: "rfc2307",
			"cache_credentials"	: "true",
			"enumerate"	: "true",
                        "ldap_id_use_start_tls" : "False",
                        "ldap_user_search_base" : nil,
                        "ldap_group_search_base": "ou=group,dc=suse,dc=cz"
		    ],
		],
	    ],
	],
	"sysconfig": $[
	    "ldap": $[
		"BASE_CONFIG_DN": nil,
		"BIND_DN": "uid=manager,dc=suse,dc=cz",
		"FILE_SERVER": "no"
	    ]
	],
	"init": $[
	    "scripts": $[
		"exists": false,
	    ],
	],
	"passwd": $[
	    "passwd": $[
		"plusline": "+",
		"pluslines": ["+"]
	    ]
	],
	"product" : $[ "features" :
	    $[ "EVMS_CONFIG" : "nazdar" ],
	],
	"target"	: $[
	    "size"	: -1,
	    "stat"	: $[1:2],
	],
    ];

    map EX = $[
	"target": $[
	    "bash": 0,
	    "bash_output": $[
		// for Pam::Enabled ("mkhomedir") call
		"stdout"	: "something:",
	    ]
	],
	"passwd": $[
	    "init": true
	]
    ];
    Testsuite::Init ([READ,$[],$[]], nil);

    import "Ldap";

    Testsuite::Test (``(Ldap::Read ()), [ READ, $[], EX ], 0);

    Testsuite::Test (``(Ldap::Export ()), [ READ, $[], EX ], 0);

}
