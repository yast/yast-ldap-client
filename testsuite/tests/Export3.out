Read	.etc.ldap_conf.v."/etc/ldap.conf"."uri" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."host" "localhost"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."base" "dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."ssl" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_cacertdir" "/etc/openldap/cacerts/"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_cacertfile" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_checkpeer" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_passwd" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_shadow" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_group" "ou=group,dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."pam_password" "crypt"
Read	.sysconfig.ldap.BASE_CONFIG_DN nil
Read	.sysconfig.ldap.FILE_SERVER "no"
Read	.sysconfig.ldap.BIND_DN "uid=manager,dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_map_attribute" 0
Execute	.passwd.init $["base_directory":"/etc"] true
Read	.passwd.passwd.pluslines ["+"]
Read	.etc.krb5_conf.v.libdefaults.default_realm ["SUSE.CZ"]
Read	.etc.krb5_conf.v."SUSE.CZ"."kdc" ["kdc.suse.cz", "kdc.suse.de"]
Read	.etc.sssd_conf.v."domain/default"."krb5_realm" "SUSE.CZ"
Read	.etc.sssd_conf.v."domain/default"."krb5_kdcip" nil
Read	.etc.sssd_conf.v."domain/default"."ldap_schema" "rfc2307"
Read	.etc.sssd_conf.v."domain/default"."cache_credentials" "true"
Read	.etc.sssd_conf.v."domain/default"."enumerate" "true"
Return	true
Return	$["base_config_dn":"", "bind_dn":"uid=manager,dc=suse,dc=cz", "create_ldap":false, "file_server":false, "krb5_kdcip":"kdc.suse.cz,kdc.suse.de", "krb5_realm":"SUSE.CZ", "ldap_domain":"dc=suse,dc=cz", "ldap_server":"localhost", "ldap_tls":false, "login_enabled":true, "member_attribute":"member", "mkhomedir":true, "nss_base_group":"ou=group,dc=suse,dc=cz", "pam_password":"crypt", "sssd":true, "sssd_cache_credentials":true, "sssd_enumerate":true, "sssd_ldap_schema":"rfc2307", "sssd_with_krb":true, "start_autofs":true, "start_ldap":true, "tls_cacertdir":"/etc/openldap/cacerts/"]
